core/src/main/scala/kafka/log/Log.scala

```diff
@@ -266,10 +266,6 @@ class Log(val dir: File,
       activeSegment.timeIndex.resize(config.maxIndexSize)
     }
 
-    /**
-     * Load the largest timestamp into memory.
-     */
-    logSegments.foreach(_.loadLargestTimestamp())
```

Comment:
The maxTimestampSoFar and offsetOfMaxTimestamp are loaded when the log segment is constructed. So it seems that we don't need to load it again.